cpp/memilio/utils/date.h

```diff
@@ -228,21 +250,19 @@ inline int get_day_in_year(Date date)
     auto year  = date.year;
     auto month = date.month;
     auto day   = date.day;
-    assert(month > 0 && month < 13 && day > 0 && day < 32);
+    assert(month > 0 && month < 13 && day > 0 && day <= get_month_length(date));
 
     if (month > 1) {
-        std::array<int, 12> month_len;
-        if (((year % 4 == 0) && (year % 100 != 0)) || (year % 400 == 0)) {
-            // leap year
-            month_len = {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+        std::array<int, 12> part_sum;
+        if (is_leap_year(year)) {
+            std::partial_sum(date.month_lengths_leap_year.begin(), date.month_lengths_leap_year.end(),
```

Comment:
same as line 211-217 :) try without code replication. When you shift this to a function you can also make the if condition more easier since its basically just the same function with slightly different args.